### PR TITLE
Add feedback request when mirrord runs 100 times (#36).

### PR DESCRIPTION
**Pull request by meowjesty**
- Issue: #36 

Adds a global counter that asks the user to leave feedback after 100 runs of mirrord, (then it resets itself?).

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -334,3 +345,26 @@ export class MirrordAPI {
         });
     }
 }
+
+
+/** 
+* Updates the global feedback counter. When it hits `FEDBACK_COUNTER_REVIEW_AFTER` mirrord runs,
```
Comment (eyalb181) [inline src/api.ts:351]:
```suggestion
* Updates the global feedback counter. When it hits `FEEDBACK_COUNTER_REVIEW_AFTER` mirrord runs, 
```

File: src/api.ts [hunk 2]
```diff
@@ -334,3 +345,26 @@ export class MirrordAPI {
...
     }
 }
+
+
+/** 
+* Updates the global feedback counter. When it hits `FEDBACK_COUNTER_REVIEW_AFTER` mirrord runs, 
+* displays a message asking the user to like mirrord.
+*/ 
+function tickFeedbackCounter() {
+    const counter = parseInt(globalContext.globalState.get(FEEDBACK_COUNTER) ?? '0') || 0;
```
Comment (eyalb181) [inline src/api.ts:355]: Why is the `|| 0` necessary?

File: src/api.ts [hunk 3]
```diff
@@ -334,3 +345,26 @@ export class MirrordAPI {
...
+    globalContext.globalState.update(FEEDBACK_COUNTER, `${counter + 1}`);
```
Comment (eyalb181) [inline src/api.ts:356]: Is the stringification necessary? Doesn't `update` stringify for you?

File: src/api.ts [hunk 4]
```diff
@@ -334,3 +345,26 @@ export class MirrordAPI {
...
+
+    if (counter >= FEEDBACK_COUNTER_REVIEW_AFTER) {
+        new NotificationBuilder()
+            .withMessage(`Enjoying mirrord? Don't forget to leave a review! Also consider giving us some feedback, we highly appreciate it!`)
```
Comment (eyalb181) [inline src/api.ts:360]:
```suggestion
            .withMessage(`Enjoying mirrord? Don't forget to leave a review! Also consider giving us some feedback, we'd highly appreciate it!`)
```

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: src/api.ts (hunk 3 above)
Comment (meowjesty) [reply src/api.ts:356]: Good question, I just copied this from the other instance where we change numbers in `globalState`, will check why it was done this way.

File: src/api.ts (hunk 2 above)
Comment (meowjesty) [reply src/api.ts:355]: Same answer as the `update` stringify question.

Comment (meowjesty) [reply src/api.ts]: Not needed as well.

**Review by eyalb181 — APPROVED** [review]

**Pull request merged**